c83764718.lua

```diff
@@ -23,7 +23,7 @@ function c83764718.target(e,tp,eg,ep,ev,re,r,rp,chk,chkc)
 end
 function c83764718.activate(e,tp,eg,ep,ev,re,r,rp)
 	local tc=Duel.GetFirstTarget()
-	if tc:IsRelateToEffect(e) and aux.NecroValleyFilter()(tc) then
+	if tc:IsRelateToEffect(e) and e:GetHandler():IsHasEffect(EFFECT_NECRO_VALLEY_CHECK) and aux.NecroValleyFilter()(tc) then
```

Comment:
_and e:GetHandler() is related to e_
Maybe getting the current effect handler and NecroValley player effect in `NecroValleyFilter` and check immunity will work? I know card immune to player effect can be weird.

Reply:
EFFECT_NECRO_VALLEY
for monsters: The monster is protected by Necrovalley.
for players: The Graveyard of the player is protected by Necrovalley.

I think this check should use a different effect code.